test/e2e/test_gcp_broker.go

```diff
@@ -0,0 +1,134 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"knative.dev/eventing/test/lib/duck"
+	"net/url"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	eventingtestlib "knative.dev/eventing/test/lib"
+	eventingtestresources "knative.dev/eventing/test/lib/resources"
+	"knative.dev/pkg/test/helpers"
+
+	// The following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
+	"github.com/google/knative-gcp/test/e2e/lib"
+	"github.com/google/knative-gcp/test/e2e/lib/resources"
+)
+
+/*
+PubSubWithBrokerTestImpl tests the following scenario:
+
+                              5                   4
+                    ------------------   --------------------
+                    |                 | |                    |
+          1         v	      2       | v         3          |
+(Sender) ---> GCP Broker ---> dummyTrigger -------> Knative Service(Receiver)
+                    |
+                    |    6                   7
+                    |-------> respTrigger -------> Service(Target)
+
+Note: the number denotes the sequence of the event that flows in this test case.
+*/
+
+func GCPBrokerTestImpl(t *testing.T, authConfig lib.AuthConfig) {
+	senderName := helpers.AppendRandomString("sender")
+	targetName := helpers.AppendRandomString("target")
+
+	client := lib.Setup(t, true, authConfig.WorkloadIdentity)
+	defer lib.TearDown(client)
+
+	// Create a target Job to receive the events.
+	makeTargetJobOrDie(client, targetName)
+
+	u := createGCPBroke(t, client, targetName)
+
+	// Just to make sure all resources are ready.
+	// Sleep for 60 seconds because of the delay of mounted ConfigMaps of GCP Broker. We will decrease the sleep time later after we update the annotation of GCP Broker which triggers an immediate refresh of mounted ConfigMaps
+	time.Sleep(60 * time.Second)
+
+	// Create a sender Job to sender the event.
+	senderJob := resources.SenderJob(senderName, []v1.EnvVar{{
+		Name:  "BROKER_URL",
+		Value: u.String(),
+	}})
+	client.CreateJobOrFail(senderJob)
+
+	// Check if dummy CloudEvent is sent out.
+	if done := jobDone(client, senderName, t); !done {
+		t.Error("dummy event wasn't sent to broker")
+		t.Failed()
+	}
+	// Check if resp CloudEvent hits the target Service.
+	if done := jobDone(client, targetName, t); !done {
+		t.Error("resp event didn't hit the target pod")
+		t.Failed()
+	}
+}
+
+func createGCPBroke(t *testing.T, client *lib.Client, targetName string) url.URL {
+	brokerName := helpers.AppendRandomString("gcp")
```

Comment:
There are some duplicate code of E2E tests of GCP Broker and Broker with PubSub Channel. I will create another issue to refactor it.